### PR TITLE
For #8043 Hook up AwesomeBarView.setOnEditSuggestionListener

### PR DESCRIPTION
**Pull request by jonalmeida**
Re-push of https://github.com/mozilla-mobile/fenix/pull/13346 with required AC version bump.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by jonalmeida** [comment]
cc: @cypressious

**Comment by cypressious** [comment]
Awesome

**Comment by jonalmeida** [comment]
I'm fixing the tests.

File: buildSrc/src/main/java/AndroidComponents.kt [hunk 1]
```diff
@@ -3,5 +3,5 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 object AndroidComponents {
-    const val VERSION = "53.0.20200804130112"
+    const val VERSION = "54.0.20200806165119"
```
Comment (eliserichards) [inline buildSrc/src/main/java/AndroidComponents.kt:6]: Let's wait to land this until the [AC update](https://github.com/mozilla-mobile/fenix/pull/13394) lands

Comment (jonalmeida) [reply buildSrc/src/main/java/AndroidComponents.kt:6]: This PR is failing for the same reasons the AC one is. I'm trying to figure out what the cause of the test failures are.

Comment (jonalmeida) [reply buildSrc/src/main/java/AndroidComponents.kt:6]: Once we figure out that out, we can use the latest version of the nightly build.

**Comment by jonalmeida** [comment]
I'm more confident now that the test failures are because of our `setSearchTerm` changes. I'm still trying to figure out what the expectation is for Fenix.

File: buildSrc/src/main/java/AndroidComponents.kt (hunk 1 above)
Comment (eliserichards) [reply buildSrc/src/main/java/AndroidComponents.kt]: 👍 awesome

**Comment by jonalmeida** [comment]
@cypressious FYI, I found that with the breaking test, this would have introduced a regression into Fenix. I fixed the issue and wrote another test to preserve the current behaviour.

**Comment by jonalmeida** [comment]
> bors r+

I make this mistake too. This is Fenix, there is no bors! 😄

**Pull request merged**